env0/resource_team_project_assignment.go

```diff
@@ -0,0 +1,91 @@
+package env0
+
+import (
+	"context"
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTeamProjectAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTeamProjectAssignmentCreateOrUpdate,
+		ReadContext:   resourceTeamProjectAssignmentRead,
+		UpdateContext: resourceTeamProjectAssignmentCreateOrUpdate,
+		DeleteContext: resourceTeamProjectAssignmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"team_id": {
+				Type:        schema.TypeString,
+				Description: "id of the team",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "id of the project",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Description: "the assigned role (Admin, Planner, Viewer, Deployer)",
+				Required:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					_ = client.Role(val.(string))
+					return
+				},
+			},
+		},
+	}
+}
+
+func resourceTeamProjectAssignmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	id := d.Id()
+	projectId := d.Get("project_id").(string)
+	assignments, err := apiClient.TeamProjectAssignments(projectId)
+
+	if err != nil {
+		return diag.Errorf("could not get TeamProjectAssignment: %v", err)
+	}
+
+	for _, assignment := range assignments {
+		if assignment.Id == id {
+			d.Set("project_id", assignment.ProjectId)
+			d.Set("team_id", assignment.TeamId)
+			d.Set("role", assignment.ProjectRole)
```

Comment:
should `break` after found?

Reply:
also, should we use `SetId` in this method?

Reply:
> should `break` after found?
yes 7992743f77155c9561457b34c80d497861bedf74
> also, should we use `SetId` in this method?

nope, we already have it ( they are equal...)